### DISCUSSION TITLE
When should the content-hash update?

### DISCUSSION THREAD

**Discussion by fwindolf** [python-poetry/poetry#8270] [Q&A]
I'm using 1.5.1 and I can run a `poetry update ...` to update my dependency. This will result in the version bumped in poetry.lock, but no update of the content-hash. If I run a `poetry add ...@latest` the version changes. 

I can't seem to find this behaviour explained in the docs, and as I build containers with lots of heavy ML libraries in CI/CD, I try to avoid downloading + installing if not necessary. This is done by checking the content-hash, as I understood it as a "hash of the content". 

Either it's not, or there is a bug and the content hash update is forgotten on `poetry update`.

**Comment by radoering** [comment]
The purpose of the content hash is to check if the lockfile is consistent with the pyproject.toml, i.e. the relevant parts of the pyproject.toml have not been changed since the lockfile was created. That's why the content hash is built from relevant parts of the pyproject.toml. When you run `poetry update` only the lockfile is changed and that's why the content hash does not change.

That's probably not explained because it's considered internals.

**Comment by vivodi** [comment]
`content-hash` means the hash of `pyproject.toml` instead of `poetry.lock`.

**Comment by tkfoss** [comment]
I found this thread when trying to generate content-hash while updating poetry lock by hand. I created a small script for that here https://gist.github.com/tkfoss/823358ba8ff2d5d4dbc1c741279b5daf 
Might be useful to someone ✌

**Reply by Secrus** [reply]
You shouldn't be updating the lock file by hand

**Reply by tkfoss** [reply]
🏴‍☠️

**Reply by Secrus** [reply]
Care to share what use case makes you edit lock file manually?

**Reply by tkfoss** [reply]
I haven't managed to set up poetry with gemfury, so now I install all the packages I can from the source with pip, and let poetry install fail on them. I have to ask someone else to recreate lock when updating dependency package, so I update the lock by hand. Thank you for asking!

for reference this works on the pipeline but for some reason doesn't work on my machines (mac and debian pc)
```
poetry --no-ansi config repositories.bigBiz https://pypi.fury.io/bigBiz 
poetry --no-ansi config http-basic.bigBiz $GEMFURY_TOKEN $GEMFURY_TOKEN
```

**Reply by Res260** [reply]
Hi @Secrus here is my usecase for changing the lock file by hand:

Internally, Pypi.org is blocked. We need to use our private nonprod pypi mirror. So I add a `tool.poetry.source` to my `pyproject.toml` and then generate a `poetry.lock`.

However, when I want to do a production deployment, I cannot reach the _nonprod_ pypi mirror, I need to reach the _prod_ pypi mirror. This does not work out of the box, because the `poetry.lock` has nonprod pypi mirror URLs in it.

I then `sed` to change the URLs, but `poetry install` still fails, because even though the packages are downloaded correctly, Poetry STILL tries to reach the nonprod mirror for some reason that I don’t know. So I need to change the `pyproject.toml` and the `content-hash` of `poetry.lock` because otherwise it doesn’t work. Poetry doesn’t seem to work for constrained corporate environments :(
